src/main.rs

```diff
@@ -247,6 +248,27 @@ impl SendState {
     }
 }
 
+fn all_filenames_from_path_names(fnames: &[String]) -> Vec<String> {
+    let mut all_files = Vec::with_capacity(fnames.len());
+
+    for fname in fnames {
+        let metadata = std::fs::metadata(fname).unwrap();
```

Comment:
For a v1 quick tool, `unwrap` is also what I would have done, but now that we are adding more features and making it a bit nicer to use, I think we should also pay a bit of attention to error reporting and make that nicer.

In particular here, if you use it as `fastsync send «addr» file1 file2`, if one of them does not exist it crashes with
```
thread 'main' panicked at src/main.rs:255:49:
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
but it doesn’t tell you which of the files does not exist. We could bring in one of these error crates or we could just add a match here:

```suggestion
        let metadata = match std::fs::metadata(fname) {
            Ok(m) => m,
            Err(err) => {
                eprintln!("Failed to access path '{fname}': {err}");
                std::process::exit(1);
            }
        };
```